researcher.lic

```diff
@@ -0,0 +1,178 @@
+      when 'summon', 'summoning', 'planes'
+        'planes'
+      when 'theurgy', 'road'
+        'road'
+    end 
+  end
+
+
+  def set_researching(status)
+    UserVars.researcher['researching'] = status
+    UserVars.researcher['timestamp'] = Time.now
+  end
+
+  def researching
+    return false if Flags['research-partial']
+    return false if Flags['research-complete']
+    return false unless DRSpells.active_spells.include?('Gauge Flow')
+    UserVars.researcher['researching']
+  end
+
+  def add_flags
+    # Use the same flags from crossing-training
+    Flags.add('research-partial', 'there is still more to learn before you arrive at a breakthrough', 'distracted by combat', 'distracted by your spellcasting', 'You lose your focus on your research project', 'you forget what you were') unless Flags['research-partial']
+    Flags.add('research-complete', '^Breakthrough!', 'You complete reviewing your knowledge of the \w+ symbiosis and commit the details to memory') unless Flags['research-complete']
+  end
+
+  def check_status
+    case DRC.bput('research status', 'You estimate that you will complete it a few minutes from now', "You're not researching anything", "You have completed .+ of a project about")
+      when /You estimate that you will complete it a few minutes from now/
+        set_researching(true)
+      when /You're not researching anything/, /You have completed .+ of a project about/
+        set_researching(false)
+    end
+  end
+
+  def check_research
+    if Flags['research-partial']
+      Flags.reset('research-partial')
+      set_researching(false)
+      start_research
+    elsif Flags['research-complete']
+      Flags.reset('research-complete')
+      set_researching(false)
+      @current_topic = nil
+    else
+      start_research
+    end
+  end
+
+  def start_research
+    return if researching
+
+    research_start_msgs = [
+      'You focus', 
+      'You tentatively', 
+      'You confidently', 
+      'Abandoning the normal', 
+      'You are already busy',
+      'You start to research',
+      'With a mixture of rational concern',
+      'You expertly coach',
+      'You begin to bend',
+      'You bring yourself',
+      'Inwardly, you make an exercise',
+      'You orient in the figurative direction'
+    ]
+
+    research_start_fail_msgs = [
+      'You cannot begin',
+      'Usage:', 
+      'You do not know how to research'
+    ]
+
+
+    check_gaf
+    @current_topic = 'stream' if @current_topic == 'attunement'
+    case bput("research #{@current_topic} 300", research_start_msgs, research_start_fail_msgs )
+    when 'You cannot begin'
+      DRCA.release_magical_research
+      start_research
+    when *research_start_msgs
+      set_researching(true)
+    when 'Usage:', 'You do not know how to research'
+      DRC.message("Invalid research option")
+      set_researching(false)
+      exit
+    end
+  end
+
+  def check_gaf
+    return if DRSpells.active_spells.include?('Gauge Flow')
```

Comment:
Should we also check its duration? What happens if GAF wears off mid-research?

Reply:
Not a bad idea, plus means that I need to learn about DRSpells.. So, bonus! :)

Will work on it here shortly